api/src/controllers/auth.ts

```diff
@@ -232,6 +233,15 @@ router.get(
 			req.session.redirect = req.query.redirect as string;
 		}
 
+		emitAsyncSafe(`oauth.${req.params.provider}`, {
+			event: `oauth.${req.params.provider}`,
+			action: 'provider',
+			schema: null,
+			payload: req.params.provider,
+			accountability: req.accountability,
+			user: null,
+		});
+
```

Comment:
Lets make this a `.before` too, and call the `action` "redirect" By making it a blocking `before`, you can prevent the particular user ip from redirecting to your oauth provider

Reply:
yes I was thinking if we could add the redirect `req.query.redirect` as part of the payload. @rijkvanzanten do you think it is needed

Reply:
I don't know if it's needed, but we might as well! Good idea to make the payload an object so it's easier later to add more things if needed (otherwise it'll be a breaking change)

Reply:
done